include/grpcpp/impl/codegen/callback_common.h

```diff
@@ -125,11 +128,28 @@ class CallbackWithSuccessTag
   // there are no tests catching the compiler warning.
   static void operator delete(void*, void*) { assert(0); }
 
-  CallbackWithSuccessTag() : call_(nullptr), ops_(nullptr) {}
+  CallbackWithSuccessTag() : call_(nullptr) {}
 
   CallbackWithSuccessTag(grpc_call* call, std::function<void(bool)> f,
-                         CompletionQueueTag* ops)
-      : call_(call), func_(std::move(f)), ops_(ops) {
+                         CompletionQueueTag* ops) {
+    Set(call, f, ops);
+  }
+
+  CallbackWithSuccessTag(const CallbackWithSuccessTag&) = delete;
+  CallbackWithSuccessTag& operator=(const CallbackWithSuccessTag&) = delete;
+
+  ~CallbackWithSuccessTag() {
+    if (call_ != nullptr) {
+      func_ = nullptr;
+      g_core_codegen_interface->grpc_call_unref(call_);
+    }
+  }
+
+  void Set(grpc_call* call, std::function<void(bool)> f,
```

Comment:
It's illegal to call Set multiple times because of the ref on call. Add that as a comment or change the function name